### PR TITLE
feat(core): add logic to save and update email template

### LINKED ISSUE #275: Add email templates
Mockups: [Figma](https://www.figma.com/design/kQOH4s5DnfWL0p6o6tlMQz/Draft-%5BATS%5D?node-id=2017-20575&t=Uz2BdPmurdxuFS46-1)

### Email templates page

- [x] Route: /settings/recruitment/email\_templates
- [x] Title: Settings \- Recruitment \- Email templates | FreeATS
- [x] There shouldn't support for personal Email templates (no Owner column, only Admins can add templates)  
- [x] Heading “Email templates”  
- [x] Place “Add email template” button in the same row with heading; set 16px spacing between them  
  - [x] On click redirect to Edit email template page  
  - [x] On Small and Extra small breakpoints place button below heading with 16px spacing between them  
- [x] Table  
  - [x] 25 items per page, as on grid pages  
  - [x] Columns: Name, Added, Updated  
    - [x] Set Added and Updated columns to take as little space as possible, so that long email template name would fit  
    - [x] Values in Name column should be Template names; on click redirect to corresponding Edit email template page  
    - [x] Sort by Updated on desc column  
  - [x] Set 16px spacing between table and pagination

### Edit email template page

- [x] Route: /settings/recruitment/email\_templates/ID
- [x] Title: Settings \- Recruitment \- Edit email template | FreeATS
- [x] Note that while this page is opened, Email templates tab in Navigational sidebar should be highlighted  
  * But when Email templates tab is clicked, redirect to Email templates page  
- [x] Heading “Edit email template”  
- [x] Email template name input:  
  * Label: Template name  
  * Placeholder: Enter template name  
- [x] Subject input:  
  * Label: Subject  
  * Placeholder: Enter email subject  
- [ ] Message text area:  
  - [x] Label: Message  
  - [x] Placeholder: Enter email message  
  - [x] Rich text  
  - [ ] If possible to implement within 4 hours:  
    * Add “{{placeholders}} button  
      * On hover show tooltip with text “Placeholders are used to add specific information to your emails without the need for manual input.”   
    * On click list all template variables (see below) in dropdown menu  
      * On hovering over template variable option display tooltip with description of variable (i.e. “first name of the recipient)  
    * Reference for implementation \- [Slack](https://toughbyte.slack.com/archives/C066V39GZPT/p1732549671726499?thread_ts=1732545943.551819&cid=C066V39GZPT)  
    * Apply our Default Icon Button style for rich text controls icon-buttons  
    * If wasn't possible to implement this variant \- write Rabbit holes  
- [x] Placeholders list:  
  - [x] Implement this variant, if wasn’t possible via “{{placeholders}}” button  
  - [x] Label: Placeholders  
  - [x] Place [help-circle](https://tabler.io/icons/icon/help-circle) icon (outline, 1.25 stroke-width; tblr-gray-500 color; 17x17px size) near label  
    - [x] on hover show tooltip with text “Placeholders are used to add specific information to your emails without the need for manual input.”  
  - [x] Add support for these template variables and list them on this page:  
    * {{first\_name}} \- first name of the recipient  
      * Use first part of the full name, space as separator between parts  
    * {{full\_name}} \- first and last name of the recipient  
    * {{sender\_first\_name}} \- first name of the person sending the email  
      * Use first part of the full name, space as separator between parts  
    * {{sender\_full\_name}} \- first and last name of the person sending the email  
    * {{company}} \- your company name  
    * {{position}} \- position name of the most recent assigned position where candidate is qualified  
- [x] “Save” button  
  * If new email template is added: when “Save” button is pressed on this page add email template to the list in table on Email templates page and display regular success alert “All changes have been saved.”  
  * If existing email template is edited: display regular success alert “All changes have been saved.”

### PR DESCRIPTION
**Pull request by wrmk**
Closes #275.

- [x] I have reviewed my code in "Files changed" tab.
- [x] I have re-read the issue and this PR fully resolves it.

### CODE REVIEW

**Comment by wrmk** [comment]
add pagination
update policy: add update? and create? actions

File: app/assets/stylesheets/src/ats/_settings.scss [hunk 1]
```diff
@@ -1,15 +1,13 @@
 #settings {
-  .edit_email_template {
```
Comment (wrmk) [inline app/assets/stylesheets/src/ats/_settings.scss:2]: form was changed from `form_for` to `form_with` and there is became no `edit_email_template`

**Review by Lyams — COMMENTED** [review]
Could you implement it so that when saving a template it moves to a message (about success/error)?

**Comment by Lyams** [comment]
https://github.com/freeats/freeats/issues/328 related issue

**Comment by Lyams** [comment]
> Could you implement it so that when saving a template it moves to a message (about success/error)?

https://github.com/freeats/freeats/issues/328 related issue

**Pull request merged**